### DISCUSSION TITLE
Should I expect greater write IO compared to zfs-auto-snapshots?

### DISCUSSION THREAD

**Discussion by deviantintegral** [zrepl/zrepl#622] [Q&A] [answered]
A few weeks ago, I converted a pool from using `zfs-auto-snapshot` for snapshots to zrepl. Everything is working fine, but I noticed that there's significantly more write activity to the disks correlated with the time I switched to zrepl. It looks like it occurs when zrepl taks snapshots. If I stop the zrepl service, IO returns to normal. `sda` is one half a a mirror vdev with a small boot pool and the root pool:

<img width="1531" alt="image" src="https://user-images.githubusercontent.com/255023/180822623-6d569af0-51d7-4ecd-a717-4900ff2706fa.png">

I think the configuration I have is very similar to what I had with `zfs-auto-snapshot`. Given the underlying zfs hasn't changed, I was surprised to see such a difference. Today I added the line omitting docker images from snapshots, and that reduced the writes, so I'm guessing whatever is driving the IO scales with the number of datasets being snapshotted.

Is this expected due to some feature of zrepl that requires lots of writes when taking snapshots, or a possible bug?

```
global:
  logging:
    # use syslog instead of stdout because it makes journald happy
    - type: syslog
      format: human
      level: warn

jobs:
- name: rpool
  type: snap
  filesystems: {
    "rpool<": true,
    "rpool/ROOT/ubuntu_ankqzy/var/lib/docker<": false,
    "rpool/lxc/images<": false,
    "rpool/lxc/deleted<": false,
  }
  snapshotting:
    type: periodic
    interval: 15m
    prefix: zrepl_
  pruning:
    keep:
      # make sure manually created snapshots by the administrator are kept
      - type: regex
        regex: "^manual_.*"
      - type: regex
        regex: "^zfs-auto-snap.*"
      - type: grid
        grid: 1x1h(keep=all) | 24x1h | 30x1d
        regex: "^zrepl_.*"

- name: bpool
  type: snap
  filesystems: {
    "bpool<": true,
  }
  snapshotting:
    type: periodic
    interval: 15m
    prefix: zrepl_
  pruning:
    keep:
      # make sure manually created snapshots by the administrator are kept
      - type: regex
        regex: "^manual_.*"
      - type: regex
        regex: "^zfs-auto-snap.*"
      - type: grid
        grid: 1x1h(keep=all) | 24x1h | 7x1d
        regex: "^zrepl_.*"
```

**Comment by problame** [comment, marked as answer]
>  Today I added the line omitting docker images from snapshots, and that reduced the writes, so I'm guessing whatever is driving the IO scales with the number of datasets being snapshotted.

That'll likely be it. Please post that graph for a time range before & after you added that line.

**Reply by deviantintegral** [reply]
I think there were some other unrelated issues amplifying IO. I've cleaned those up, and even with re-enabling the docker zfs volumes IO is now roughly the same as with zfs-auto-snapshot. Thank you!
